tests/zero_temperature/statistics

```diff
@@ -3,13 +3,12 @@
 # 3: Number of mesh cells
 # 4: Number of Stokes degrees of freedom
 # 5: Number of temperature degrees of freedom
-# 6: Iterations for temperature solver
-# 7: Iterations for Stokes solver
-# 8: Velocity iterations in Stokes preconditioner
-# 9: Schur complement iterations in Stokes preconditioner
-# 10: Time step size (seconds)
-# 11: Minimal temperature (K)
-# 12: Average temperature (K)
-# 13: Maximal temperature (K)
-0 0.000000000000e+00 1024 9539 4225 0 37 50  8 1.000000000000e+00 0.00000000e+00 0.00000000e+00 0.00000000e+00 
-1 1.000000000000e+00 1024 9539 4225 0 29 30 30 3.141683944776e+15 0.00000000e+00 0.00000000e+00 0.00000000e+00 
+# 6: Iterations for Stokes solver
+# 7: Velocity iterations in Stokes preconditioner
```

Comment:
What happened in the original pull request was that as soon as a field changed from being skipped to being solved, all of the lines in the statistics file had the column with the number of iterations (they were zero for the time steps where solving was skipped). 
However, the fields appear in the file in the order the output is generated, so this would lead to cases where the number of temperature/composition iterations were not in their normal places, but moved to a different column. 
Because of that, we now generate the output with the number of iterations even if the equation is not solved.